### PR TITLE
Implement testing by cypress & enzyme

### PR DESCRIPTION
**Pull request by dylanjongbloed**

### CODE REVIEW

File: cypress/integration/TimeEntries.js [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+  it('finds the content "type"', function() {
+    cy.visit('http://localhost:8000/')
+
+    cy.contains('Time Entries')
+
+    cy.get('#activity')
+      .type('fake activity')
+      .should('have.value', 'fake activity')
+  })
+})
```
Comment (RubenWerdmuller) [inline cypress/integration/TimeEntries.js]: Check on line endings through the changed files.

File: src/components/menu-toggle/__tests__/MenuToggle.test.tsx [hunk 2]
```diff
@@ -0,0 +1,14 @@
+import * as React from 'react';
+import { shallow } from 'enzyme';
+
+import MenuToggle from '../MenuToggle';
+
+it('matches snapshot', () => {
+  const menuToggle = shallow(
+    <MenuToggle
+    toggleMenu={() => null}
+    isMenuVisible= {true}
```
Comment (RubenWerdmuller) [inline src/components/menu-toggle/__tests__/MenuToggle.test.tsx]: Sort alphabetically.

File: src/components/team-member-form/__tests__/TeamMemberForm.test.tsx [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+import { shallow } from 'enzyme';
+
+import TeamMemberForm from '../TeamMemberForm';
+
+it('matches snapshot', () => {
+  const teamMemberForm = shallow(
+    <TeamMemberForm
+      isTeamMemberFormVisible={true}
+      createTeamMember={({}) => {}}
+      toggleTeamMemberForm={() => null}
```
Comment (RubenWerdmuller) [inline src/components/team-member-form/__tests__/TeamMemberForm.test.tsx]: Sort props alphabetically.

File: src/components/team-members/__tests__/TeamMembers.test.tsx [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+  lastName: 'string',
+  locality: 'string',
+  postalCode: 'string',
+  startingDate: 'string'
+}];
+
+it('matches snapshot', () => {
+  const teamMembers = shallow(
+    <TeamMembers
+      teamMembers ={mockedTeamMembers}
```
Comment (RubenWerdmuller) [inline src/components/team-members/__tests__/TeamMembers.test.tsx]: Sort props alphabetically.

File: src/components/time-entry-form/TimeEntryForm.tsx [hunk 5]
```diff
@@ -38,12 +38,12 @@ function TimeEntryForm({
     });
   };
 
-  const handleCancel = (event): void => {
+  const handleCancel = (): void => {
     event.preventDefault();
```
Comment (RubenWerdmuller) [inline src/components/time-entry-form/TimeEntryForm.tsx]: Is `event.PreventDefault()` needed here? It doesn't seem like a form is being submitted.

File: src/components/time-entry/__tests__/TimeEntry.test.tsx [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+it('matches snapshot', () => {
+  const timeEntry = shallow(
+    <TimeEntry
+      id={123}
+      client={{
+        branch: '123',
+        clientName: '123',
+        id: 123,
+        locality: '123',
+        postalCode: '123',
```
Comment (RubenWerdmuller) [inline src/components/time-entry/__tests__/TimeEntry.test.tsx]: Sort props alphabetically.

**Review by RubenWerdmuller — APPROVED** [review]

File: src/components/time-entry-form/TimeEntryForm.tsx (hunk 5 above)
Comment (dylanjongbloed) [reply src/components/time-entry-form/TimeEntryForm.tsx]: correcto!

File: src/services/__tests__/time-entries-api.test.js [hunk 7]
```diff
@@ -0,0 +1,33 @@
...
+    postalCode: '123'
+  },
+  id: 123,
+  startTimestamp: 'bla',
+  stopTimestamp: 'bla'
+}]
+
+it('fetches TimeEntries from the server', () => {
+  const mockSuccessResponse = {mockedTimeEntries};
+  const mockJsonPromise = Promise.resolve(mockSuccessResponse);
```
Comment (RobbinHabermehl) [inline src/services/__tests__/time-entries-api.test.js]: Are you aware that the `mockSuccessResponse` looks like this:

```
{
  mockedTimeEntries: [
    {
      // properties
    }
  ]
}
```
That doesn't seem to make sense. Why don't you just resolve like this: `Promise.resolve(mockedTimeEntries)`.

**Review by RobbinHabermehl — CHANGES_REQUESTED** [review]

File: package.json [hunk 8]
```diff
@@ -14,7 +14,7 @@
     "start": "npm run develop & npm run api",
     "storybook": "start-storybook -p 6006 -s ./static",
     "test": "TZ=Europe/Amsterdam LANG=nl_NL.UTF-8 LC_ALL=nl_NL.UTF-8 jest",
-    "open cypress": "cypress open"
+    "open cypress": " npx cypress open"
```
Comment (RobbinHabermehl) [inline package.json]: See my previous comment. Also, remove the leading space.

**Review by RobbinHabermehl — CHANGES_REQUESTED** [review]

File: index.d.ts [hunk 9]
```diff
@@ -1,2 +1,5 @@
+/// <reference types="typescript" />
 declare module '*.svg';
 declare module '*.css';
+declare module '*.jpg';
+declare module 'styled-components';
```
[the remark this answers is not in the document]
Comment (dylanjongbloed) [reply index.d.ts]: I added this after the terminal suggested me to do so upon testing components that use shared components (in this case an input)

File: src/components/time-entry-heading/__tests__/TimeEntryHeading.test.tsx [hunk 10]
```diff
@@ -0,0 +1,13 @@
+import * as React from 'react';
+import { shallow } from 'enzyme';
+
+import TimeEntryHeading from '../TimeEntryHeading';
+
+it('matches snapshot', () => {
+  const timeEntryHeading = shallow(
+    <TimeEntryHeading
+    startTimestamp='test'
```
[the remark this answers is not in the document]
Comment (dylanjongbloed) [reply src/components/time-entry-heading/__tests__/TimeEntryHeading.test.tsx]: adjusted!

File: src/services/__tests__/time-entries-api.test.js (hunk 7 above)
Comment (dylanjongbloed) [reply src/services/__tests__/time-entries-api.test.js]: you're right! adjusted

**Review by RobbinHabermehl — APPROVED** [review]

**Pull request merged**